specs/Accessibility.md

```diff
@@ -0,0 +1,78 @@
+
+## Focus
+- Whenever possible, focus should land only on elements that have concrete implicit or explicit ARIA role (`<button />`, `role='button'`, `role='menuitem'`, ....)
+- onClick event handlers need to be handled on focusable elements with implicit or explicit ARIA role and not on their parent/child elements
+- When reacting on focus or on hover screen reader interaction needs to be specified. Some screen readers do not have concept of hovering and not always focus elements, so there needs to be another way for the screen reader users to interact with such elements. For example showing popups on focus or on hover needs to be justified and alternative interaction compatible with the screen readers need to be provided
+
+## Screen reader support
+User needs to be able to interact with the application using:
+- screen reader with virtual cursor navigation (browse mode)
+- screen reader without virtual cursor navigation (application mode)
+
+All requirements for keyboard navigation are valid in both of these screen reader use cases. User needs to be able to understand which element is focused and how he can interact with it just based on the information provided by the screen reader.
+
+Following combinations of clients and screen readers is supported/required:
+- Windows/Chrome - JAWS, NVDA
+- Windows/Edge - Narrator
+- Windows/Firefox - NVDA
+- macOS/Chrome - VoiceOver
+
+## Contrast
+- There has to be sufficient contrast between background and foreground of the component (text, icon...) in all component states and themes.
+
+## Zoom
+- Elements need to be rendered correctly when zoomed up to 400%
+
+# Components and Behaviors
+There are two areas that come together to achieve accessibility:
+* Components - implemented for each supported framework (currently React), when rendered, need to be [semantically correct](https://en.wikipedia.org/wiki/Semantic_HTML)
+* Behaviors - framework independent, intended to add ARIA roles, ARIA attributes and keyboard handling on top of the components based on their type and state. In the future, behaviors development will move from the [Stardust react](https://github.com/stardust-ui/react) repo to [Stardust accessibility](https://github.com/stardust-ui/accessibility) repo.
+
+**Accessibility behaviors** reflect the need to have a set of validated and carefully tested alternatives available for the consumers that guarantee the accessibility of the component in various use cases. In some cases, accessibility can be highly oppinionated and the behaviors allow consumers to implement accessibility aspects of the components in a customized way, that better fits the requirements of the consumer.
+
+Accessibility behavior is a function that takes properties of the component combined with state and maps it to:
+- `role`, `aria-*` attributes, `tabIndex` and other attributes related to accessibility
```

Comment:
I think there should be guidance added somewhere about how to add/modify ARIA props. For example, when I was looking at how to add support for `aria-mixed` (indeterminate state for checkbox), I looked at the contributing.md file for Accessibility and couldn't find anything about it. @layershifter pointed me to packages/accessibility/src/behaviors/Checkbox/checkboxBehavior.ts and until he did that it was unclear that this is how Fluent UI's accessibility infra is structured - that there's a whole other package.